### ISSUE TITLE
Bug: Inconsistent error auth/invalid-email between createUser and signIn

### ISSUE THREAD

**Issue by RobertHeim** [firebase/firebase-admin-java#263]
### [REQUIRED] Step 2: Describe your environment

  * Operating System version: Ubuntu 14.04.5 LTS
  * Firebase SDK version: Java firebase SDK com.google.firebase:firebase-admin:6.8.0
  * Library version: JS firebase: "^5.3.1"
  * Firebase Product: auth

### [REQUIRED] Step 3: Describe the problem

Trying to login with a strange email such as `ł€¶ŧł@example.com` results in `auth/invalid-email`. So I would expect that creating a user with this email address would throw the same error up-front, but it does not.

#### Steps to reproduce:
Create a user with this email `ł€¶ŧł@example.com`. It does *not* throw an FirebaseAuthException with code `auth/invalid-email`.

#### Relevant Code:

```
String email = "ł€¶ŧł@example.com"; 
try {
    CreateRequest request = new CreateRequest()
        .setEmail(email)
        .setPassword(pw);
    return FirebaseAuth.getInstance().createUser(request);
} catch (FirebaseAuthException e) {
  String msgCode = e.getErrorCode();
  // expected msgCode to be "auth/invalid-email", but the exception does not occur
}
```
In contrast, after creating the user through the backend / java sdk without an error, the frontend uses the firebase javascript library to login, but it fails with `auth/invalid-email`:
```
firebase.auth().signInWithEmailAndPassword(email, pw)
  .catch((e) => {
    console.log(e.code);
   // this does show "auth/invalid-email"
  };
```

**Comment by hiranya911** [comment]
@bojeil-google, looks like the Admin SDK allows the email address, but the client SDK does not.

**Comment by bojeil-google** [comment]
Client validator is quite lax: https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth/src/utils.js#L484
Something else is throwing the error.

**Comment by wti806** [comment]
@RobertHeim  Please update the SDK version you are using. This issue was fixed in [v5.7.1](https://firebase.google.com/support/release-notes/js#5.7.1). Thanks!
